gravitee-apim-console-webui/src/management/api/portal/subscriptions/components/dialogs/creation/api-portal-subscription-creation-dialog.component.spec.ts

```diff
@@ -207,6 +214,198 @@ describe('Subscription creation dialog', () => {
         expect(await harness.isCustomApiKeyInputDisplayed()).toBeFalsy();
       });
     });
+    describe('With custom apiKey enabled and shared api key enabled', () => {
+      beforeEach(() => {
+        TestBed.configureTestingModule({
+          declarations: [TestComponent],
+          imports: [ApiPortalSubscriptionsModule, NoopAnimationsModule, GioHttpTestingModule, MatIconTestingModule],
+          providers: [
+            { provide: SubscriptionService },
+            {
+              provide: InteractivityChecker,
+              useValue: {
+                isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+                isTabbable: () => true, // This traps tabbable checks and so avoid warnings when dealing with
+              },
+            },
+            {
+              provide: 'Constants',
+              useFactory: () => {
+                const constants = CONSTANTS_TESTING;
+                set(constants, 'env.settings.plan.security', {
+                  customApiKey: { enabled: true },
+                  sharedApiKey: { enabled: true },
+                });
+                return constants;
+              },
+            },
+          ],
+        });
+        fixture = TestBed.createComponent(TestComponent);
+        httpTestingController = TestBed.inject(HttpTestingController);
+        fixture.detectChanges();
+        component = fixture.componentInstance;
+        loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+      });
+
+      afterEach(() => {
+        jest.clearAllMocks();
+        httpTestingController.verify();
+      });
+
+      it('should not be able to select apikey mode when no subscription and custom api key', async () => {
```

Comment:
the title does not fit the content of the test

Reply:
good catch